assets/src/block-editor/QueryBlockExtension/QueryBlockExtension.js

```diff
@@ -1,154 +1,125 @@
+                  for (const group of document.querySelectorAll('.components-toolbar-group')) {
+                    // Find the views group from toolbar
+                    if (group.innerHTML.includes('List view') || group.innerHTML.includes('Grid view')) {
+                      group.style.display = 'none';
                     }
                   }
-                }
-
-                // Move the carousel button to the views group if exists
-                if (carouselBtn && group.contains(carouselBtn) && viewsGroup) {
-                  viewsGroup.append(group);
-                }
+                }, 10);
+
+                return () => {
+                  clearTimeout(timeout);
+                };
+            }
+          }, [isPostTemplate]);
+
+          useEffect(() => {
+            if(selectedBlock && selectedBlock.name === 'core/post-template') {
+              const parentBlock = wp.data.select('core/block-editor')
+                .getBlock(wp.data.select('core/block-editor')
+                .getBlockParents(selectedBlock.clientId)[0]);
+
+              if(parentBlock && parentBlock.attributes.namespace === 'planet4-blocks/actions-list') {
+                setPostTemplate(selectedBlock);
               }
-            }, 50);
-
-            return () => {
-              clearTimeout(timeout);
-            };
-          }
-        }, [props.isSelected]);
+            }
+          }, [selectedBlock]);
+
+          useEffect(() => {
+            // Reset every time a new block is selected
+            setPostTemplate(null)
+            setSelectedBlock(wp.data.select('core/block-editor').getSelectedBlock());
+          }, [
+            wp.data.select('core/block-editor').getSelectedBlock(),
+          ]);
+
+          return useMemo(() => (
+            <>
+              <InspectorControls>
+                <PanelBody
+                  title={__('Layout view', 'planet4-blocks-backend')}
+                  initialOpen={true}
+                >
+                  {attributes.layout && (
+                    <>
+                      <RadioControl
+                        label={__('Display as', 'planet4-blocks-backend')}
+                        help={__('The layout of the action|post list', 'planet4-blocks-backend')}
+                        selected={attributes.layout.type}
+                        options={layoutTypes}
+                        onChange={type => onChangeHandler(type)}
+                      />
+                    </>
+                  )}
+                </PanelBody>
+              </InspectorControls>
+              <BlockEdit {...props} />
+            </>
+          ), [
+            attributes,
+            isPostTemplate,
+            selectedBlock,
+            wp.data.select('core/block-editor').getSelectedBlock()
+          ]);
+        }
 
-        return useMemo(() => (
-          <>
-            {isSelectedBlock('planet4-blocks/actions-list') && (
```

Comment:
Don't need to check this anymore since is being checked [here](https://github.com/greenpeace/planet4-master-theme/pull/2163/commits/21580b351f947d2bd483460aaa03ec5144c6ef65#diff-def48ca20835d6ea3428e5ac7aed2b8002e9f59bbb236f97c0976c8f1aacd79eR24)